api-helpers/sitemap.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  getArticlesPaginationForGrid,
+  getArticlesPaginationForList,
+  getArticlesSlugs,
+} from './articles';
+import { closeConnection, openConnection } from './db';
+
+type Item = {
+  readonly path: string;
+  readonly changefreq: 'daily' | 'monthly' | 'always' | 'hourly' | 'weekly' | 'yearly' | 'never';
+  readonly priority: number;
+};
+
+const staticItems: readonly Item[] = [
+  { path: '/', changefreq: 'hourly', priority: 1 },
+  { path: '/o-serwisie', changefreq: 'monthly', priority: 0.5 },
+  { path: '/zglos-serwis', changefreq: 'monthly', priority: 0.5 },
+
+  { path: '/list', changefreq: 'hourly', priority: 0.9 },
+  { path: '/grid', changefreq: 'hourly', priority: 0.9 },
+];
+
+export async function getSitemap() {
+  try {
+    const prisma = await openConnection();
+
+    const [gridCursors, listCursors, articleSlugs] = await Promise.all([
+      getArticlesPaginationForGrid(prisma),
+      getArticlesPaginationForList(prisma),
+      getArticlesSlugs(prisma),
+    ]);
+
+    const dynamicItems: readonly Item[] = [
+      ...articleSlugs.map(({ slug }) => ({
+        path: `/artykuly/${slug}`,
+        changefreq: 'monthly' as const,
+        priority: 0.5,
+      })),
+      ...gridCursors.map((cursor) => ({
+        path: `/grid/${cursor}`,
+        changefreq: 'hourly' as const,
+        priority: 0.4,
+      })),
+      ...listCursors.map((cursor) => ({
+        path: `/list/${cursor}`,
+        changefreq: 'hourly' as const,
+        priority: 0.4,
+      })),
+    ];
+
+    return sitemapXml([...staticItems, ...dynamicItems]);
+  } finally {
+    await closeConnection();
+  }
```

Comment:
co w przypadku wystąpienia błędu?

Reply:
A co powinniśmy zrobić?

Reply:
no właśnie się zastanawiam, co w przypadku jak baza zwróci jakis błąd zamiast danych co wtedy zwróci funkcja? obiekt błędu zamiast tablicy danych?

Może powinno się jakoś ten błąd w catchu wyłapać i coś z nim zrobić

Reply:
Jeśli będzie błąd, którego nie przewidzieliśmy, to API zwróci 500.